supervisor/container.go

```diff
@@ -226,30 +236,41 @@ func (c *Container) addProcess(processId, stdin, stdout, stderr string, spec *sp
 	c.ownerPod.Processes[processId] = p
 	c.Processes[processId] = p
 
-	e := Event{
-		ID:        c.Id,
-		Type:      EventProcessStart,
-		Timestamp: time.Now(),
-		PID:       processId,
-	}
-	c.ownerPod.sv.Events.notifySubscribers(e)
+	// Wait for the VM to acknowledge the process has attached to avoid a
+	// libcontainer deadlock.
+	resultCh := make(chan error)
 
 	go func() {
 		c.ownerPod.podStatus.AddExec(c.Id, processId, "", spec.Terminal)
 		err := c.ownerPod.vm.AddProcess(c.Id, processId, spec.Terminal, spec.Args, spec.Env, spec.Cwd, p.stdio)
```

Comment:
This was motivated by it being unclear what the expectation higher up containerd is if AddProcess fails. If we fail to ever start the process, then an attempt at recovering containers later isn't going to find it since it was probably a runv/hyperstart error. So better not to notify it was added until we're sure it was.